src/reformat_libyuv.c

```diff
@@ -350,429 +368,136 @@ avifResult avifImageYUVToRGBLibYUV8bpc(const avifImage * image,
+    avifBool isYVU = lutIsYVU[rgb->format];
+    const struct YuvConstants * matrix = isYVU ? matrixYVU : matrixYUV;
+    if (image->yuvFormat == AVIF_PIXEL_FORMAT_YUV400) {
+        if (yuv400ToRgbMatrix != NULL) {
+            if (yuv400ToRgbMatrix(image->yuvPlanes[AVIF_CHAN_Y],
                                   image->yuvRowBytes[AVIF_CHAN_Y],
-                                  image->yuvPlanes[AVIF_CHAN_U],
-                                  image->yuvRowBytes[AVIF_CHAN_U],
-                                  image->yuvPlanes[AVIF_CHAN_V],
-                                  image->yuvRowBytes[AVIF_CHAN_V],
                                   rgb->pixels,
                                   rgb->rowBytes,
-                                  matrixYUV,
+                                  matrix,
                                   image->width,
                                   image->height) != 0) {
                 return AVIF_RESULT_REFORMAT_FAILED;
             }
             return AVIF_RESULT_OK;
         }
-    } else if (rgb->format == AVIF_RGB_FORMAT_RGB_565) {
-        // AVIF_RGB_FORMAT_BGR   *ToRGB565Matrix  matrixYUV
-
-        if (image->yuvFormat == AVIF_PIXEL_FORMAT_YUV422) {
-#if LIBYUV_VERSION >= 1838
-            if (I422ToRGB565Matrix(image->yuvPlanes[AVIF_CHAN_Y],
-                                   image->yuvRowBytes[AVIF_CHAN_Y],
-                                   image->yuvPlanes[AVIF_CHAN_U],
-                                   image->yuvRowBytes[AVIF_CHAN_U],
-                                   image->yuvPlanes[AVIF_CHAN_V],
-                                   image->yuvRowBytes[AVIF_CHAN_V],
-                                   rgb->pixels,
-                                   rgb->rowBytes,
-                                   matrixYUV,
-                                   image->width,
-                                   image->height) != 0) {
-                return AVIF_RESULT_REFORMAT_FAILED;
-            }
-            return AVIF_RESULT_OK;
-#endif
-        } else if (image->yuvFormat == AVIF_PIXEL_FORMAT_YUV420) {
-            if (I420ToRGB565Matrix(image->yuvPlanes[AVIF_CHAN_Y],
-                                   image->yuvRowBytes[AVIF_CHAN_Y],
-                                   image->yuvPlanes[AVIF_CHAN_U],
-                                   image->yuvRowBytes[AVIF_CHAN_U],
-                                   image->yuvPlanes[AVIF_CHAN_V],
-                                   image->yuvRowBytes[AVIF_CHAN_V],
-                                   rgb->pixels,
-                                   rgb->rowBytes,
-                                   matrixYUV,
-                                   image->width,
-                                   image->height) != 0) {
+    } else if (yuvToRgbMatrixFilter != NULL || yuvToRgbMatrix != NULL) {
+        int uPlaneIndex = isYVU ? AVIF_CHAN_V : AVIF_CHAN_U;
+        int vPlaneIndex = isYVU ? AVIF_CHAN_U : AVIF_CHAN_V;
```

Comment:
Nit/optional: Might be more accurate to name tese variables `plane1Index` and `plane2Index` or `firstChromaIndex` and `secondChromaIndex`. But I understand what they mean.